lib/config/config.go

```diff
@@ -162,6 +159,38 @@ type Configuration struct {
 	OriginalVersion int               `xml:"-" json:"-"` // The version we read from disk, before any conversion
 }
 
+func (cfg Configuration) GUIs() []GUIConfiguration {
+	if override := os.Getenv("STGUIADDRESSES"); override != "" {
+		// This value may be a comma separated list of urls.
+		//
+		// Each url may be of the form "scheme://address:port" or just
+		// "address:port". We need to chop off the scheme. We try to
+		// parse it as an URL if it contains a slash. If that fails,
+		// return it as is and let some other error handling handle
+		// it.
+		var overrideGUIs []GUIConfiguration
+		for _, overrideEntry := range strings.Split(override, ",") {
+			overrideGUIs = append(overrideGUIs, GUIConfigFromString(overrideEntry))
+		}
+		return overrideGUIs
+	} else if override := os.Getenv("STGUIADDRESS"); override != "" {
+		// Legacy overriding form which only supports one address.
+		return []GUIConfiguration{GUIConfigFromString(override)}
+	}
```

Comment:
As far as I understand it before only the address was set by the env var, all other GUI settings were still taken from the config file. I don't really see how to do that with a variable amount of GUIs in a sensible way.
One option would be to match given addresses with the list of GUI configs and use default values if there are more addresses given than there are GUIs in the config. That seems a bit random though. Another thing would be to just use default values when the env var is used. Honestly I don't really know the use case for this, so I am fishing in the dark.

Reply:
This is overridden by wrappers, instead of passing in the command line I think.
Use TLS and other crap had assessors would check the env var, so I suspect this is still ok.

Reply:
What I mean is that e.g. `apiService.getHandler` gets the config from a `cfg.Wrapper` (as everything else I assume) and that queries this function, which will return a default gui config except for the address which is set by the env var. Previously it got the "normal" gui config and got the address by calling `GUIConfiguration.Address` which respected the env var. Thus options like user and password were preserved when using the env var, which they aren't anymore.

Reply:
Ok, that's a fair argument, but I can't think of how to handle this properly.
Perhaps we should support `http(s)?://user:pass@host.com`. I think there is no way to set the API key from env var?

Reply:
Regardless what we do, this is going to end up a breaking change, so perhaps we just remove STGUIADDRESS, leave STGUIADDRESSESS that supports specifying user:pass.

Reply:
Fine for me. I will just wait a definitive confirmation to move forward in this direction. I am just wondering whether we really have to remove `STGUIADDRESS`. We could also extend the supported syntax for this legacy environment variable by adding support for the `user:pass` form.

Reply:
@calmh agree?
This changes gui config from object to array in the json world anyway, which is a breaking change, so I think we can axe that old syntax.

Reply:
I think the STGUIADDRESS variable is intended for wrappers and similar things mostly? As such I think it's only used/useful to say "I want to be able to access the GUI/API *here*" and I have a hard time imagining a situation where being able to specify multiple would make sense. I would just keep STGUIADDRESS and have it be in addition to whatever else is configured in the config. 

The API key can currently be set with STGUIAPIKEY. I think it makes sense to be able to specify that from the outside, again for wrappers. I'm not sure of the use case of being able to specify username and password from an environment variable.

Reply:
My point is that this is a breaking change anyway, so I don't see a reason why we could not remove the old thing and add a new thing that does a bit more, such as extracts username and password from the URL, and supports multiple. Not saying that it will be used for multiple listeners, but the ability to specify that does not do any damage.

Reply:
Yeah, we can keep the old thing for compatability,but to me it seems like a good opportunity to deprecate now, rather than later.

Reply:
Assuming we want to keep compatibility, I still don't see a consistent way to do this yet:
Before: All GUI config parameters are taken from config except url/port and key if given by env.
Now: Url/port and key can still be set by env, but then all other values are default, regardless of what is in the config.
To have the same/similar behaviour we could for example just combine the values given with the parameters in the first GUI config. That is close to what it was before, but seems a bit random in case of multiple GUI configs (what designates the "first" config). Actually that might not even be consistent, as order in json is random, if I remember correctly. I don't know whether this matters at all for those who actually use this env vars, I am just pointing out the change in behaviour which doesn't seem to be on the radar.